inference-engine/src/transformations/src/transformations/hswish_decomposition.cpp

```diff
@@ -0,0 +1,44 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/hswish_decomposition.hpp"
+
+#include <memory>
+
+#include <ngraph/opsets/opset4.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+
+ngraph::pass::HSwishDecomposition::HSwishDecomposition() {
+    // Decomposes HSwish(x) op into sub-graph x * (min(Relu(x + 3), 6) * const(1/6)
+    auto hswish = ngraph::pattern::wrap_type<opset4::HSwish>();
+
+    ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher &m) {
+        auto &pattern_to_output = m.get_pattern_value_map();
+        auto hswish_node = std::dynamic_pointer_cast<ngraph::opset4::HSwish>(pattern_to_output.at(hswish).get_node_shared_ptr());
```

Comment:
BTW you don't need to cast it to HSwish as you use only its inputs

Reply:
Changed